### PR TITLE
Allow Cylc Review to access Cylc 8 Workflow data.

### PR DESCRIPTION
**Pull request by wxtim**
These changes partially address #4113 - Depending upon discussion on that issue, this may be a complete solution.

- [x] Fix access to task jobs list. At this stage I have an informally tested fix. I've put it up for review so that reviewers can try it.
- [x] Allow access to `cylc.flow` file. Modified the list of files so that this can be a 1-stop-shop.
- [x] Existing tests for `cylc-review`  don't break differently to target branch. 😢 
- [ ] ~Create a formal test that Cylc8 Workflows can be read.~ Manual testing only because this a complex test for legacy software.
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Appropriate change log entry included.
- [x] No dependency changes.

### CODE REVIEW

**Review by MetRonnie — APPROVED** [review]
LGTM assuming it has at least been tested manually

**Comment by wxtim** [comment]
I'm not convinced that it's going to be particularly straightforward to automate a test for this. I think to test it manually that you should:
- Run at least 1 (but not necessarily the same) suite in Cylc7 and Cylc8.
- With cylc review open and browse around.
- Confirm that links and pages work.

**Comment by wxtim** [comment]
@dpmatthews  said it worked for him on a quick test.

**Review by datamel — APPROVED** [review]
I have played a cylc 8 workflow and tested cylc review both fails on master and works as expected on this branch. 
Thanks for getting me up and running with this @wxtim.
I have sent you a quick pm with the extra files from the cylc-install work.

File: lib/cylc/review_dao.py [hunk 1]
```diff
@@ -255,6 +278,21 @@ def get_suite_job_entries(
...
 
+    def is_cylc8(self, user_name, suite_name):
+        # Detemine Cylc version for a given suite: Database changes require
+        # A different database query for Cylc8.
+        suite_info = self._db_exec(
+            user_name, suite_name, 'SELECT * FROM suite_params', []
+        )
+        for row in suite_info:
+            if row[0] == u'cylc_version':
+                cylc_version = row[1]
```
Comment (oliver-sanders) [inline lib/cylc/review_dao.py:289]: Should set `cylc_version = None` before this loop just incase.

File: lib/cylc/review_dao.py [hunk 2]
```diff
@@ -255,6 +278,21 @@ def get_suite_job_entries(
...
+
+        if cylc_version[0] == '8':
```
Comment (oliver-sanders) [inline lib/cylc/review_dao.py:291]: For safety:

```suggestion
        if cylc_version and cylc_version[0] == '8':
```

**Review by oliver-sanders — COMMENTED** [review]
Code looks good, just need to test...

**Review by oliver-sanders — APPROVED** [review]

**Comment by oliver-sanders** [comment]
Tested, everything working as expected.

**Pull request merged**